web/client-api/src/main/java/io/deephaven/web/client/api/JsTable.java

```diff
@@ -1200,61 +1200,23 @@ public Promise<JsTable> snapshot(JsTable baseTable, @JsOptional Boolean doInitia
                 .then(state -> Promise.resolve(new JsTable(workerConnection, state)));
     }
 
-    /**
-     * @deprecated a promise that will be resolved with a newly created table holding the results of the join operation.
-     *             The last parameter is optional, and if not specified or empty, all columns from the right table will
-     *             be added to the output. Callers are responsible for ensuring that there are no duplicates - a match
-     *             pair can be passed instead of a name to specify the new name for the column. Supported `joinType`
-     *             values (consult Deephaven's "Joining Data from Multiple Tables for more detail): "Join" <a href=
-     *             'https://docs.deephaven.io/latest/Content/writeQueries/tableOperations/joins.htm#Joining_Data_from_Multiple_Tables'>Joining_Data_from_Multiple_Tables</a>
-     *             "Natural" "AJ" "ReverseAJ" "ExactJoin" "LeftJoin"
-     * @param joinType
-     * @param rightTable
-     * @param columnsToMatch
-     * @param columnsToAdd
-     * @param asOfMatchRule
-     * @return Promise of dh.Table
-     */
     @Override
     @JsMethod
-    @Deprecated
-    public Promise<JsTable> join(Object joinType, JoinableTable rightTable, JsArray<String> columnsToMatch,
-            @JsOptional @JsNullable JsArray<String> columnsToAdd, @JsOptional @JsNullable Object asOfMatchRule) {
-        if (joinType.equals("AJ") || joinType.equals("RAJ")) {
-            return asOfJoin(rightTable, columnsToMatch, columnsToAdd, (String) asOfMatchRule);
-        } else if (joinType.equals("CROSS_JOIN")) {
+    public Promise<JsTable> join(String joinType, JoinableTable rightTable, JsArray<String> columnsToMatch,
```

Comment:
Should the deprecation tag be removed from this? It's on the interface

Annoyingly, TS passes through the docstring from the interface if an implementation does not have its own, but does not pass through the deprecation (this is probably technically correct though). Using `table.join` has no strikethrough. If you add the deprecation tag here, then it strikes through `join`, but doesn't pull through the param info. Only pulls the summary

Reply:
Unless I misunderstand, the only way to mark something as deprecated is with jsdoc `@deprecated` - so if we mark JsTable.join() as /** @deprecated */ it naturally drops the otherwise-inherited docs... 

I'll try [`@inheritDoc`](https://typedoc.org/tags/inheritDoc/) to see if it makes these get along. We can see about tweaking the tooling to auto-inherit in cases where an annotation should be inherited...

Reply:
Yes your understanding is right. `@deprecated` isn't inherited which is technically correct because we could be deprecating its usage in the interface, but maintaining it for an implementation.

I just tried `@inheritDoc` again and looks like you can do just `@inheritDoc` and let TS infer it, or `@inheritDoc JoinableTable.join`

This seems to be right in VSCode (strike through, full hover docs). It has duplicated deprecation tags (the inherited at the top and additional at the end), but it's the best option so far

```
@deprecated
@inheritDoc
```